nav2_controller/src/nav2_controller.cpp

```diff
@@ -51,27 +52,35 @@ DwbController::DwbController()
     }, costmap_ros_);
 }
 
-DwbController::~DwbController()
+Nav2Controller::~Nav2Controller()
 {
   RCLCPP_INFO(get_logger(), "Destroying");
   costmap_executor_.cancel();
   costmap_thread_->join();
 }
 
 nav2_util::CallbackReturn
-DwbController::on_configure(const rclcpp_lifecycle::State & state)
+Nav2Controller::on_configure(const rclcpp_lifecycle::State & state)
 {
-  RCLCPP_INFO(get_logger(), "Configuring");
+  std::string local_planner_name;
+
+  RCLCPP_INFO(get_logger(), "Configuring local controller interface");
 
   costmap_ros_->on_configure(state);
 
   auto node = shared_from_this();
 
   progress_checker_ = std::make_unique<ProgressChecker>(rclcpp_node_);
 
-  planner_ = std::make_unique<dwb_core::DWBLocalPlanner>(
-    node, costmap_ros_->getTfBuffer(), costmap_ros_);
-  planner_->on_configure(state);
+  try {
+    std::string local_controller_plugin;
+    get_parameter("local_controller_plugin", local_controller_plugin);
+    local_planner_ = lp_loader_.createUniqueInstance(local_controller_plugin);
+    RCLCPP_INFO(get_logger(), "Created local controller");
```

Comment:
The INFO should contain the `local_controller_plugin` string to know what's been loaded